npm install fail on OSX with node 4.1: "error: thread-local storage is not supported "

`curl -fsSL https://goo.gl/getgrpc | bash -s nodejs`

then run `npm install`:

```
./src/core/support/cpu_posix.c:45:8: error: thread-local storage is not supported for the current target
static __thread char magic_thread_local;
       ^
1 error generated.
make: *** [Release/obj.target/gpr/src/core/support/cpu_posix.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/zengxian/workspace/grpc
gyp ERR! node -v v4.0.0
gyp ERR! node-gyp -v v3.0.1
gyp ERR! not ok

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.0.0
npm ERR! npm  v2.14.2
npm ERR! code ELIFECYCLE
npm ERR! grpc@0.11.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the grpc@0.11.1 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the grpc package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls grpc
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/zengxian/workspace/grpc/examples/node/npm-debug.log
```

--- a/binding.gyp
+++ b/binding.gyp
@@ -73,16 +73,6 @@
          ]
         ]
       }],
-      ['OS=="mac"', {
-        'xcode_settings': {
-          'MACOSX_DEPLOYMENT_TARGET': '10.9',
-          'OTHER_CFLAGS': [
-            '-fno-strict-aliasing',
-            '-std=c++11',
-            '-stdlib=libc++'
-          ]
-        }
-      }]
     ]
   },
   'targets': [
@@ -133,6 +123,16 @@
         'src/core/support/time_precise.c',
         'src/core/support/time_win32.c',
         'src/core/support/tls_pthread.c',
+      ],
+      "conditions": [
+        ['OS == "mac"', {
+          'xcode_settings': {
+            'MACOSX_DEPLOYMENT_TARGET': '10.9',
+            'OTHER_CFLAGS': [
+              '-stdlib=libc++'
+            ]
+          }
+        }]
       ],
     },
     {
@@ -291,6 +291,16 @@
         'src/core/census/initialize.c',
         'src/core/census/operation.c',
         'src/core/census/tracing.c',
+      ],
+      "conditions": [
+        ['OS == "mac"', {
+          'xcode_settings': {
+            'MACOSX_DEPLOYMENT_TARGET': '10.9',
+            'OTHER_CFLAGS': [
+              '-stdlib=libc++'
+            ]
+          }
+        }]
       ],
     },
     {

--- a/binding.gyp
+++ b/binding.gyp
@@ -73,6 +73,16 @@
          ]
         ]
       }],
+      ['OS=="mac"', {
+        'xcode_settings': {
+          'MACOSX_DEPLOYMENT_TARGET': '10.9',
+          'OTHER_CFLAGS': [
+            '-fno-strict-aliasing',
+            '-std=c++11',
+            '-stdlib=libc++'
+          ]
+        }
+      }]
     ]
   },
   'targets': [

--- a/binding.gyp
+++ b/binding.gyp
@@ -137,10 +137,7 @@
       "conditions": [
         ['OS == "mac"', {
           'xcode_settings': {
-            'MACOSX_DEPLOYMENT_TARGET': '10.9',
-            'OTHER_CFLAGS': [
-              '-stdlib=libc++'
-            ]
+            'MACOSX_DEPLOYMENT_TARGET': '10.9'
           }
         }]
       ],
@@ -305,10 +302,7 @@
       "conditions": [
         ['OS == "mac"', {
           'xcode_settings': {
-            'MACOSX_DEPLOYMENT_TARGET': '10.9',
-            'OTHER_CFLAGS': [
-              '-stdlib=libc++'
-            ]
+            'MACOSX_DEPLOYMENT_TARGET': '10.9'
           }
         }]
       ],

--- a/binding.gyp
+++ b/binding.gyp
@@ -54,7 +54,8 @@
     ],
     'include_dirs': [
       '.',
-      'include'
+      'include',
+      '<(node_root_dir)/deps/openssl/openssl/include'
     ],
     'conditions': [
       ['OS != "win"', {
@@ -73,6 +74,15 @@
          ]
         ]
       }],
+      ["target_arch=='ia32'", {
+          "include_dirs": [ "<(node_root_dir)/deps/openssl/config/piii" ]
+      }],
+      ["target_arch=='x64'", {
+          "include_dirs": [ "<(node_root_dir)/deps/openssl/config/k8" ]
+      }],
+      ["target_arch=='arm'", {
+          "include_dirs": [ "<(node_root_dir)/deps/openssl/config/arm" ]
+      }]
     ]
   },
   'targets': [

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -75,16 +75,6 @@
            ]
           ]
         }],
-        ['OS=="mac"', {
-          'xcode_settings': {
-            'MACOSX_DEPLOYMENT_TARGET': '10.9',
-            'OTHER_CFLAGS': [
-              '-fno-strict-aliasing',
-              '-std=c++11',
-              '-stdlib=libc++'
-            ]
-          }
-        }]
       ]
     },
     'targets': [
@@ -103,6 +93,16 @@
           % for source in lib.src:
           '${source}',
           % endfor
+        ],
+        "conditions": [
+          ['OS == "mac"', {
+            'xcode_settings': {
+              'MACOSX_DEPLOYMENT_TARGET': '10.9',
+              'OTHER_CFLAGS': [
+                '-stdlib=libc++'
+              ]
+            }
+          }]
         ],
       },
       % endif

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -75,6 +75,16 @@
            ]
           ]
         }],
+        ['OS=="mac"', {
+          'xcode_settings': {
+            'MACOSX_DEPLOYMENT_TARGET': '10.9',
+            'OTHER_CFLAGS': [
+              '-fno-strict-aliasing',
+              '-std=c++11',
+              '-stdlib=libc++'
+            ]
+          }
+        }]
       ]
     },
     'targets': [

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -56,7 +56,8 @@
       ],
       'include_dirs': [
         '.',
-        'include'
+        'include',
+        '<(node_root_dir)/deps/openssl/openssl/include'
       ],
       'conditions': [
         ['OS != "win"', {
@@ -75,6 +76,15 @@
            ]
           ]
         }],
+        ["target_arch=='ia32'", {
+            "include_dirs": [ "<(node_root_dir)/deps/openssl/config/piii" ]
+        }],
+        ["target_arch=='x64'", {
+            "include_dirs": [ "<(node_root_dir)/deps/openssl/config/k8" ]
+        }],
+        ["target_arch=='arm'", {
+            "include_dirs": [ "<(node_root_dir)/deps/openssl/config/arm" ]
+        }]
       ]
     },
     'targets': [

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -107,10 +107,7 @@
         "conditions": [
           ['OS == "mac"', {
             'xcode_settings': {
-              'MACOSX_DEPLOYMENT_TARGET': '10.9',
-              'OTHER_CFLAGS': [
-                '-stdlib=libc++'
-              ]
+              'MACOSX_DEPLOYMENT_TARGET': '10.9'
             }
           }]
         ],

